pkg/api/norman/customization/cluster/actions_kubeconfig.go

```diff
@@ -64,26 +65,28 @@ func (a ActionHandler) GenerateKubeconfigActionHandler(actionName string, action
 	}
 
 	if endpointEnabled {
-		clusterName := apiContext.ID
-		clusterClient, err := a.ClusterManager.UserContextNoControllers(clusterName)
-		if err != nil {
-			return err
-		}
-
-		if tokenKey != "" {
-			tokenName, tokenValue := tokens.SplitTokenParts(tokenKey)
-			// a lister is not used here because the token was recently created, therefore the lister would likely miss
-			token, err := a.TokenClient.Get(tokenName, metav1.GetOptions{})
-			if err != nil {
-				return err
-			}
-			clusterAuthToken, err := common.NewClusterAuthToken(token, tokenValue)
+		if features.TokenHashing.Enabled() {
+			clusterName := apiContext.ID
+			clusterClient, err := a.ClusterManager.UserContextNoControllers(clusterName)
 			if err != nil {
 				return err
 			}
 
-			if _, err = clusterClient.Cluster.ClusterAuthTokens("cattle-system").Create(clusterAuthToken); err != nil {
-				return err
+			if tokenKey != "" {
+				tokenName, tokenValue := tokens.SplitTokenParts(tokenKey)
+				// a lister is not used here because the token was recently created, therefore the lister would likely miss
+				token, err := a.TokenClient.Get(tokenName, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				clusterAuthToken, err := common.NewClusterAuthToken(token, tokenValue)
+				if err != nil {
+					return err
+				}
+
+				if _, err = clusterClient.Cluster.ClusterAuthTokens("cattle-system").Create(clusterAuthToken); err != nil {
+					return err
+				}
```

Comment:
Can you put this into a single function for readability. Can do L75-89 instead if preferred.

Reply:
Updated.